### PR TITLE
[WIP] try fix font find bug with skia

### LINKED ISSUE #13585: Can't change FontFamily or change no work
```
<TextBlock
    FontFamily="{Binding Font}"
    FontSize="{Binding Size}"
    Foreground="{Binding FontColor}"
    Text="{Binding Content}"
    TextElement.FontFamily="{Binding Font}" />
```

```
public FontFamily Font
{
    get
    {
        return ElementInfo.Font ?? FontFamily.DefaultFontFamilyName;
    }
    set
    {
        ElementInfo.Font = value.Name;
        OnPropertyChanged(nameof(Font));
        GetNewMargin();
    }
}
```

but use may type is same.

_Originally posted by @Coloryr in https://github.com/AvaloniaUI/Avalonia/discussions/13584_

### PR DESCRIPTION
**Pull request by Coloryr**
#13585

## What does the pull request do?
try fix font find bug

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation

## Fixed issues
Fixes #13585 
Fixes #13462

### CODE REVIEW

**Comment by Coloryr** [comment]
If font was found, the name can be not same

**Comment by timunie** [comment]
add keyword "Fixes " for fixed issues to close automatically. :-)

**Comment by Gillibald** [comment]
Please provide a failing test otherwise this can't be merged

**Comment by robloo** [comment]
@Coloryr Using FontForge you can probably create two very small fonts with maybe only one character glyph for a test.

**Comment by Coloryr** [comment]
> @Coloryr Using FontForge you can probably create two very small fonts with maybe only one character glyph for a test.

Ok, that is good way.

**Comment by Coloryr** [comment]
> Please provide a failing test otherwise this can't be merged

I create a font name is `测试` `test`, could you help me write a test?
[testtt.zip](https://github.com/AvaloniaUI/Avalonia/files/13361248/testtt.zip)

**Comment by timunie** [comment]
@Coloryr Maybe you can check out the tests we have for reference. I guess the test should go here: 
- https://github.com/AvaloniaUI/Avalonia/blob/master/tests/Avalonia.Base.UnitTests/Media/FontManagerTests.cs
- or here https://github.com/AvaloniaUI/Avalonia/blob/master/tests/Avalonia.Base.UnitTests/Media/FontFamilyTests.cs

**Comment by Coloryr** [comment]
> @Coloryr Maybe you can check out the tests we have for reference. I guess the test should go here:
> 
> * https://github.com/AvaloniaUI/Avalonia/blob/master/tests/Avalonia.Base.UnitTests/Media/FontManagerTests.cs
> * or here https://github.com/AvaloniaUI/Avalonia/blob/master/tests/Avalonia.Base.UnitTests/Media/FontFamilyTests.cs

 No, I don't know why need to the `failing test`, or what need `failing`

**Comment by timunie** [comment]
the idea is: 
1. Add a test for it without your fix. This test will fail and thus reproduce the bug you saw. 
2. Then you add your commits and run the same test again. This time the test will pass. 
3. If later on any changes happen on the font manager, we can ensure your issue isn't introduced again.

**Comment by Coloryr** [comment]
> the idea is:
> 
> 1. Add a test for it without your fix. This test will fail and thus reproduce the bug you saw.
> 2. Then you add your commits and run the same test again. This time the test will pass.
> 3. If later on any changes happen on the font manager, we can ensure your issue isn't introduced again.

Ok, I will try to write a test

**Comment by Coloryr** [comment]
> the idea is:
> 
> 1. Add a test for it without your fix. This test will fail and thus reproduce the bug you saw.
> 2. Then you add your commits and run the same test again. This time the test will pass.
> 3. If later on any changes happen on the font manager, we can ensure your issue isn't introduced again.

I don't know how to implement `without your fix`, it can add to test without git?

**Comment by timunie** [comment]
you can run the test on master branch first and it should fail. then add it to your branch and it should pass.

**Comment by Coloryr** [comment]
I don't know how to write a test

**Comment by Coloryr** [comment]
```
        [Fact]
        public void Should_Get_GlyphTypeface_With_Chinese()
        {
            using (UnitTestApplication.Start(TestServices.MockPlatformRenderInterface
               .With(fontManagerImpl: new FontManagerImpl())))
            {
                var fontFamily = new FontFamily("微软雅黑");
                var typeface = new Typeface(fontFamily);
                Assert.True(FontManager.Current.TryGetGlyphTypeface(typeface, out var glyphTypeface));
                FontManager.Current.TryGetGlyphTypeface(typeface, out var other);
                Assert.NotSame(glyphTypeface, other);
            }
        }
```
this test can test chinese font, but need install font `微软雅黑` in system

**Comment by Coloryr** [comment]
Could help me write a test?

**Comment by Gillibald** [comment]
```
 [Win32Fact("Only run on Windows")]
        public void Should_Get_SystemFont_With_Localized_Name()
        {
            using (UnitTestApplication.Start(TestServices.MockPlatformRenderInterface.With(fontManagerImpl: new FontManagerImpl())))
            {
                using (AvaloniaLocator.EnterScope())
                {
                    Assert.True(FontManager.Current.TryGetGlyphTypeface(new Typeface("微软雅黑"), out var glyphTypeface));

                    Assert.Equal("Microsoft YaHei", glyphTypeface.FamilyName);
                }
            }
        }
```

**Comment by Coloryr** [comment]
> ```
>  [Win32Fact("Only run on Windows")]
>         public void Should_Get_SystemFont_With_Localized_Name()
>         {
>             using (UnitTestApplication.Start(TestServices.MockPlatformRenderInterface.With(fontManagerImpl: new FontManagerImpl())))
>             {
>                 using (AvaloniaLocator.EnterScope())
>                 {
>                     Assert.True(FontManager.Current.TryGetGlyphTypeface(new Typeface("微软雅黑"), out var glyphTypeface));
> 
>                     Assert.Equal("Microsoft YaHei", glyphTypeface.FamilyName);
>                 }
>             }
>         }
> ```

Thank you.

File: tests/Avalonia.Skia.UnitTests/Media/FontManagerTests.cs [hunk 1]
```diff
@@ -13,6 +14,20 @@ public class FontManagerTests
     {
         private static string s_fontUri = "resm:Avalonia.Skia.UnitTests.Assets?assembly=Avalonia.Skia.UnitTests#Noto Mono";
 
+        [Win32Fact("Only run on Windows")]
```
Comment (maxkatz6) [inline tests/Avalonia.Skia.UnitTests/Media/FontManagerTests.cs:17]: Win32Fact is defined in a different assembly.
For now, you can copy Win32Fact.cs to this assembly...

**Review by Gillibald — APPROVED** [review]

**Comment by Gillibald** [comment]
Having a look on what is failing on macOS

**Comment by Coloryr** [comment]
```
##[error][xUnit.net 00:00:09.88]     Avalonia.Skia.UnitTests.Media.FontManagerTests.Should_Get_Nearest_Match_For_Custom_SystemFont [FAIL]
00:52:17 [ERR] [xUnit.net 00:00:09.88]     Avalonia.Skia.UnitTests.Media.FontManagerTests.Should_Get_Nearest_Match_For_Custom_SystemFont [FAIL]
00:52:17 [DBG]   Failed Avalonia.Skia.UnitTests.Media.FontManagerTests.Should_Get_Nearest_Match_For_Custom_SystemFont [37 ms]
00:52:17 [DBG]   Error Message:
00:52:17 [DBG]    Assert.Equal() Failure
00:52:17 [DBG]           ↓ (pos 0)
00:52:17 [DBG] Expected: Noto Mono
00:52:17 [DBG] Actual:   Helvetica
00:52:17 [DBG]           ↑ (pos 0)
00:52:17 [DBG]   Stack Trace:
00:52:17 [DBG]      at Avalonia.Skia.UnitTests.Media.FontManagerTests.Should_Get_Nearest_Match_For_Custom_SystemFont() in /Users/runner/work/1/s/tests/Avalonia.Skia.UnitTests/Media/FontManagerTests.cs:line 277
00:52:17 [DBG] Results File: /Users/runner/work/1/s/artifacts/test-results/_Mac-1702600275320_2023-12-15_00_52_10.trx
00:52:17 [DBG] 
00:52:17 [DBG] Failed!  - Failed:     1, Passed:   175, Skipped:     2, Total:   178, Duration: 7 s - Avalonia.Skia.UnitTests.dll (net6.0)
```
I don't have a Mac computer, so i can't possible to debug and track the issue.
But think is actually `Skia` issue
image

**Pull request closed without merging**